pubtools/_pulp/task.py

```diff
@@ -143,3 +146,28 @@ def main(self):
 
         self.run()
         return 0
+
+
+class CDNCached(FastPurgeClientService):
+    # provding CDN cache related features
+
+    def _flush_cdn(self, repos):
```

Comment:
- Should this perhaps be added to FastPurgeClientService class instead?  I guess in 99% of cases where we are getting a fastpurge client, we intend to use it to flush cache of some repos.  If you want to stick with introducing a separate class, could you please put that in a new file rather than stick it in the main task.py?
- I don't think it should be named with leading `_` - you certainly intend this method to be called from outside of this class.  Why give it a private name... what is the `_` supposed to communicate here?
- Should the `@step` decorator perhaps also be applied once here, rather than having to wrap the method in each place it's called?

Reply:
Idea here was to define a class of tasks that interact with CDN cache i.e. the tasks that make any modifications to CDN cache are task of type CDNCached. So added this class to task.py. The method with _ is available to flush CDN cache but be used within the definition/method specific to the task.
However, looking at the comments above, it might be better to have this as a generic class rather that being narrowed down to a task type. In that case, I'll put it in a new file with public method and step decorator.